src/include/daos_obj.h

```diff
@@ -347,6 +347,52 @@ daos_obj_generate_id(daos_obj_id_t *oid, daos_ofeat_t ofeats,
 	oid->hi |= hdr;
 }
 
+/** Flags for oclass hints */
+enum {
+	/** Flags to control OC Redundancy */
+	DAOS_OC_RDD_DEF		= (1 << 0),	/** Default - use RF prop */
+	DAOS_OC_RDD_NO		= (1 << 1),	/** No redundancy */
+	DAOS_OC_RDD_RP		= (1 << 2),	/** Replication */
+	DAOS_OC_RDD_EC		= (1 << 3),	/** Erasure Code */
+	/** Flags to control OC Sharding */
+	DAOS_OC_SHD_DEF		= (1 << 4),	/** Default - use 1 grp */
+	DAOS_OC_SHD_TINY	= (1 << 5),	/** <= 4 grps */
+	DAOS_OC_SHD_REG		= (1 << 6),	/** max(128, 25%) */
+	DAOS_OC_SHD_HI		= (1 << 7),	/** max(256, 50%) */
+	DAOS_OC_SHD_EXT		= (1 << 8),	/** max(1024, 80%) */
+	DAOS_OC_SHD_MAX		= (1 << 9),	/** 100% */
+};
+
+/**
+ * Generate a DAOS object ID by encoding the private DAOS bits of the object
+ * address space. This allows the user to either select an object class
+ * manually, or ask DAOS to generate one based on some hints provided.
+ *
+ * \param[in]	coh	Container open handle.
+ * \param[in,out]
+ *		oid	[in]: Object ID with low 96 bits set and unique inside
+ *			the container.
+ *			[out]: Fully populated DAOS object identifier with the
+ *			the low 96 bits untouched and the DAOS private bits
+ *			(the high 32 bits) encoded.
+ * \param[in]	ofeats	Feature bits specific to object
+ * \param[in]	cid	Class Identifier.
+ *			If set to 0 (unknown), and no hints flags are specified,
+ *			we use the container RF property to select the
+ *			redundancy level, and assume a tiny object for now.
+ *			If one of the MAX GRP oclass is set (SX, RP_2/3GX, etc.)
+ *			the oclass chosen will be modified to have X represent
+ *			the closest GRP available in the predefined oclass list.
```

Comment:
Sounds good, although i would probably be less specific in the actual doxygen comment and claim that passing a cid is for advanced users and class will be otherwise generated from hint.